service/gcs/main.go

```diff
@@ -39,13 +39,22 @@ func memoryLogFormat(metrics *cgroupstats.Metrics) logrus.Fields {
 func readMemoryEvents(efdFile *os.File, cgName string, threshold int64, cg cgroups.Cgroup) {
 	// Buffer must be >= 8 bytes for eventfd reads
 	// http://man7.org/linux/man-pages/man2/eventfd.2.html
+	triggered := false
 	buf := make([]byte, 8)
 	for {
 		if _, err := efdFile.Read(buf); err != nil {
 			logrus.WithError(err).WithField("cgroup", cgName).Error("failed to read from eventfd")
 			return
 		}
 
+		_, err := os.Lstat(fmt.Sprintf("/sys/fs/cgroup/memory/%s/cgroup.event_control", cgName))
```

Comment:
What's the point of checking if this file exists? It's not clear from the comment below.

Reply:
https://github.com/opencontainers/runc/blob/e5f2eae5a56b89a801e1482b0d079051acca97f2/libcontainer/notify_linux.go#L55 Found this from runc. I don't think we're actually even seeing this event being sent as otherwise you'd think for N pods launched we'd have at minimum N events in our logs but thats not the case. Mostly just to be safe. Would you like it explained more?

Reply:
What I get from that is that you're checking if `cgroup.event_control` exists as a way to check if the cgroup is being torn down? I didn't feel that was explained by the comment below. I would move the comment up above the `Lstat` line, and rephrase it (e.g. `Sometimes an event is sent during cgroup teardown, but does not indicate that the threshold was actually crossed. In the teardown case the cgroup.event_control file won't exist anymore, so check that to determine if we should ignore this event.).

Reply:
Changing, thats much better